blockstore/blockstore.go

```diff
@@ -37,9 +37,21 @@ type BlockstoreIterator interface {
 
 // BlockstoreGC is a trait for blockstores that support online garbage collection
 type BlockstoreGC interface {
-	CollectGarbage() error
+	CollectGarbage(options map[interface{}]interface{}) error
```

Comment:
The difficulty is that the functional option would be defined in the base type and used in a subtype.
I guess we could make a `GCConfig` object and apply there so that we avoid the parsing.
So yeah, I'll change it to use functional options, it's cleaner.

Reply:
Basically, we have two cases:

1. This doesn't generalize. Why are we pretending it does?
2. It does generalize. So generalize it.

One-off abstractions are just a pain.

Reply:
yup, I respect your strong feelings on the matter!

Reply:
done, functional options it is!